pkg/controller/scale_up.go

```diff
@@ -19,27 +19,28 @@ func (c *Controller) ScaleUp(opts scaleOpts) (int, error) {
 		return untainted, err
 	}
 
-	// remove the number of nodes that were just untainted and the remaining is
-	// how much to increase the cloud provider node group by
+	// remove the number of nodes that were just untainted and the remaining is how much to increase the cloud provider node group by
 	opts.nodesDelta -= untainted
 
-	// check that untainting the nodes doesn't do bring us over max nodes
-	if opts.nodesDelta <= 0 {
-		log.Warnf("Scale up delta is less than or equal to 0 after clamping: %v. Will not scale up cloud provider.", opts.nodesDelta)
-		return 0, nil
-	}
+	if opts.nodesDelta > 0 {
+		// check that untainting the nodes doesn't do bring us over max nodes
+		if opts.nodesDelta <= 0 {
```

Comment:
issue: this can never be hit, is this the correct logic?

Reply:
I reverted it back to the original implementation, but yeah that can never be hit. I'll refactor to an implementation which is equivalent